invenio_users_resources/records/api.py

```diff
@@ -214,13 +216,25 @@ def avatar_color(self):
     @classmethod
     def create(cls, data, id_=None, validator=None, format_checker=None, **kwargs):
         """Create a new User and store it in the database."""
-        # NOTE: we don't use an actual database table, and as such can't
-        #       use db.session.add(record.model)
-        with db.session.begin_nested():
-            # create_user() will already take care of creating the profile
-            # for us, if it's specified in the data
-            user = current_datastore.create_user(**data)
-            return cls.from_model(user)
+        try:
+            # Check if email and  username already exists by another account.
+            errors = {}
+            existing_email = User.query.filter_by(email=data["email"]).first()
```

Comment:
compatibility: could you please use `db.session.query(User).filter_by(...)` instead of `User.query.filter_by`. the `db.session.query(User)` works with sqlalchemy <2.0 too but `User.query` doesn't work anymore in sqlalchemy >= 2.0